packages/tpetra/core/example/BlockCrs/driver.cpp

```diff
@@ -0,0 +1,536 @@
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Michael A. Heroux (maherou@sandia.gov)
+//
+// ************************************************************************
+// @HEADER
+#include <cmath>
+#include <iostream>
+#include <iomanip>
+#include <sstream>
+
+#include <Tpetra_DefaultPlatform.hpp>
+#include <Tpetra_Version.hpp>
+
+#include <MatrixMarket_Tpetra.hpp>
+
+#include <Teuchos_RCP.hpp>
+#include <Teuchos_GlobalMPISession.hpp>
+#include <Teuchos_FancyOStream.hpp>
+
+#include "typedefs.hpp"
+#include "utils.hpp"
+#include "MeshDatabase.hpp"
+
+//#define PRINT_VERBOSE 1
+
+using namespace BlockCrsTest;
+
+int main (int argc, char *argv[]) 
+{
+  using Teuchos::RCP;
+  using Teuchos::TimeMonitor;
+
+  const auto INVALID_GLOBAL_ORDINAL = Teuchos::OrdinalTraits<global_ordinal_type>::invalid();
+
+  auto out = Teuchos::getFancyOStream (Teuchos::rcpFromRef (std::cout));
+  
+  // MPI boilerplate
+  Teuchos::GlobalMPISession mpiSession (&argc, &argv, NULL);
+  RCP<const Teuchos::Comm<int> > comm = Tpetra::DefaultPlatform::getDefaultPlatform ().getComm();
+
+  // Command-line input
+  LO num_global_elements_i = 2, num_global_elements_j = 2, num_global_elements_k = 2;
+  LO num_procs_i = 1, num_procs_j = 1, num_procs_k = 1;
+  LO blocksize = 5, nrhs = 1, repeat = 100;
+
+  Teuchos::CommandLineProcessor clp(false);
+  clp.setDocString("BlockCrs performance test using 3D 7 point stencil.\n");
+  clp.setOption("num-elements-i", &num_global_elements_i, "Number of cells in the I dimension.");
+  clp.setOption("num-elements-j", &num_global_elements_j, "Number of cells in the J dimension.");
+  clp.setOption("num-elements-k", &num_global_elements_k, "Number of cells in the K dimension.");
+  clp.setOption("num-procs-i", &num_procs_i, "Processor grid of (npi,npj,npk); npi*npj*npk should be equal to the number of MPI ranks.");
+  clp.setOption("num-procs-j", &num_procs_j, "Processor grid of (npi,npj,npk); npi*npj*npk should be equal to the number of MPI ranks.");
+  clp.setOption("num-procs-k", &num_procs_k, "Processor grid of (npi,npj,npk); npi*npj*npk should be equal to the number of MPI ranks.");
+  clp.setOption("blocksize", &blocksize, "Block size. The # of DOFs coupled in a multiphysics flow problem.");
+  clp.setOption("nrhs", &nrhs, "Number of right hand sides to solve for.");
+  clp.setOption("repeat", &repeat, "Number of iterations of matvec operations to measure performance.");
+
+  {
+    auto r_parse = clp.parse(argc, argv, comm->getRank() == 0 ? &std::cout : NULL);
+    if (r_parse == Teuchos::CommandLineProcessor::PARSE_HELP_PRINTED) return 0;
+    if (r_parse != Teuchos::CommandLineProcessor::PARSE_SUCCESSFUL  ) return -1;
+
+    TEUCHOS_ASSERT(num_procs_i*num_procs_j*num_procs_k == comm->getSize());
+  }
+  
+  // Initialize Kokkos
+  Kokkos::initialize();
```

Comment:
Please please pass in the command-line arguments to Kokkos::initialize!  Tpetra::initialize would also take care of this for you.

Reply:
"--help" option print is not guarded in kokkos. This prints out multiple kokkos help descriptions. 
This should be fixed.

Reply:
Also all warning messages of OMP environement variable setup prints multiple times. Is there a simple way to avoid these multiple prints ?